controllers/virtualmachine/virtualmachine_controller.go

```diff
@@ -341,6 +361,9 @@ func (r *Reconciler) ReconcileNormal(ctx *pkgctx.VirtualMachineContext) (reterr
 	// Add this VM to prober manager if ReconcileNormal succeeds.
 	r.Prober.AddToProberManager(ctx.VM)
 
+	// Upgrade schema fields where needed
+	upgradeSchema(ctx)
```

Comment:
I wonder if we should call this function before `CreateOrUpdateVirtualMachine()` so that the VM is reconciled with the correct upgraded spec?

Reply:
Seems to work as expected either way, but moved to calling it before. I had originally done it after, since `Status.BiosUUID` is set in calls that happen during CreateOrUpdateVirtualMachine()

Reply:
That field should already be set for existing VMs. As our internal upgrade script is executed before a new VMOP controller is applied and reconciles VMs, calling this function at the beginning also simulates the same process. Thanks for addressing this.